sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/client/traits/AmqpConfigTrait.java

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.amqp.client.traits;
```

Comment:
This is an interesting problem. Amqp types are defined in azure-core-amqp package, so this trait needs to leave here. When I attempted to use the same package name like for other traits - `com.azure.core.client.traits` - I got a warning that package is already defined elsewhere. Should we leave this as is or attempt to use same package name as azure-core does? (if it's even possible).

@JonathanGiles @srnagar @lmolkova @alzimmermsft what do you think?

Reply:
We should definitely not use the same package. That is not allowed under the Java 9 module spec (no split packages). We should introduce an amqp specific package if there needs to be traits in that library.